File: src/convert2Lambert.py
#!/usr/bin/env python

import rospy
from functools import partial
import pyproj

from geometry_msgs.msg import Twist
from geometry_msgs.msg import Vector3
from geometry_msgs.msg import Vector3Stamped
from sensor_msgs.msg import NavSatFix
from sensor_msgs.msg import Imu
from tf.transformations import euler_from_quaternion, quaternion_from_euler

latitude 	= 0.0
longitude 	= 0.0
latitude_mission 	= 0.0
longitude_mission 	= 0.0

roll = pitch = yaw = 0.0

def gpsCallback(msg):
	global latitude, longitude

	latitude   = msg.latitude
	longitude  = msg.longitude

def angleCallback(msg):
	global yaw
	
	yaw = msg.vector.z

def convert():
	global latitude, longitude, longitude_mission, latitude_mission

	PROJECT = partial(
	pyproj.transform,
	pyproj.Proj(init='epsg:4326'),
	pyproj.Proj(init='epsg:2154'))

	x_lambert, y_lambert = PROJECT(longitude, latitude)
	x_mission_lambert, y_mission_lambert = PROJECT(longitude_mission, latitude_mission)

	return x_lambert, y_lambert, x_mission_lambert, y_mission_lambert

rospy.init_node('convert2Lambert') 

input_GPS_msg 		= rospy.get_param('Input_GPS_msg', 'nav')
input_yaw_msg 		= rospy.get_param('Input_yaw_msg', 'imu_attitude')


output_msg 			= rospy.get_param('Output_msg', 'gps_angle_boat')

sub = rospy.Subscriber(input_mission_msg, NavSatFix, missionCallback)
sub = rospy.Subscriber(input_GPS_msg, NavSatFix, gpsCallback) 
sub = rospy.Subscriber(input_yaw_msg, Vector3Stamped, angleCallback) 
pub = rospy.Publisher(output_msg, Twist, queue_size=10)

rate = rospy.Rate(25) 

pose = Twist()

while not rospy.is_shutdown(): 

	x_lambert, y_lambert, x_mission_lambert, y_mission_lambert = convert()

	pose = Twist(Vector3(x_lambert, y_lambert, yaw), Vector3(0, 0, 0))

	pub.publish(pose) 

	rate.sleep() 

File: doodles/convert2Lambert.cpp
#include <iostream>
#include <string>
#include <cmath>

#include "ros/ros.h"
#include "sensor_msgs/NavSatFix.h"
#include "geometry_msgs/Pose.h"

using namespace std;

double latitude, longitude;
double x_lambert, y_lambert;

void chatCallback(const sensor_msgs::NavSatFix::ConstPtr& msg)
{
    latitude   = msg->latitude;
    longitude  = msg->longitude;
}

void convert()
{
	x_lambert = latitude;
	y_lambert = longitude;
}

int main(int argc, char** argv)
{
	string input_msg;
	string output_msg;

        // Ros init

    ros::init(argc, argv, "convert2Lambert");
    ros::NodeHandle n;

    ros::Rate loop_rate(25);

        // Initials parameters

    n.param<string>("Input_msg", 	input_msg, 	"unknown");
    n.param<string>("Output_msg", 	output_msg, "unknown");

        // Subscribe

    ros::Subscriber input_sub = n.subscribe(input_msg, 1000, chatCallback);

        // Publish

    ros::Publisher output_pub = n.advertise<geometry_msgs::Pose>(output_msg, 1000);
    geometry_msgs::Pose lambert_msg;
    
    
    while(ros::ok())
    {      
    	convert();
    	
    	lambert_msg.position.x = x_lambert;
    	lambert_msg.position.y = y_lambert;

    	output_pub.publish(lambert_msg);
    	
        ros::spinOnce();
        loop_rate.sleep();
    }

    return EXIT_SUCCESS;
}